src/coord/rmr/rmr.c

```diff
@@ -47,6 +48,10 @@ static MRCluster *cluster_g = NULL;
 // Number of shards in the cluster (main-thread variable)
 extern size_t NumShards;
 
+// Local node ID (set from main-thread when topology is updated, and may be accessed from worker
+// thread upon replying to a query - hence it is synchronized reference counting)
+static NodeIdRef *local_node_id_g = NULL;
```

Comment:
Consider making it static and not heap allocated

Reply:
Will still need to initialize the lock somewhere, so was easier keeping it a pointer.